docs-content/getting-started/fullstack-frameworks/next-js/3_client.md

```diff
@@ -0,0 +1,214 @@
+---
+title: Next.js Client
+slug: client
+heading: Next.js Client Instrumentation
+createdAt: 2023-10-03T00:00:00.000Z
+updatedAt: 2023-10-03T00:00:00.000Z
+---
+
+## Installation
+
+```shell
+# with npm
+npm install @highlight-run/next
+```
+
+## Client Instrumentation
+
+This sections adds session replay and frontend error monitoring to Highlight. This implementation requires React 17 or greater. If you're behind on React versions, follow our [React.js docs](../../3_client-sdk/1_reactjs.md)
+
+1. If you're using Page Router, you'll want to add `HighlightInit` to `_app.tsx`.
+
+```jsx
+// pages/_app.tsx
+import { AppProps } from 'next/app'
+import CONSTANTS from '../app/constants'
+import { HighlightInit } from '@highlight-run/next/client'
+
+export default function MyApp({ Component, pageProps }: AppProps) {
+	return (
+		<>
+			<HighlightInit
+				// excludedHostnames={['localhost']}
+				projectId={CONSTANTS.NEXT_PUBLIC_HIGHLIGHT_PROJECT_ID}
+				serviceName="my-nextjs-frontend"
+				tracingOrigins
+				networkRecording={{
+					enabled: true,
+					recordHeadersAndBody: true
+				}}
+			/>
+
+			<Component {...pageProps} />
+		</>
+	)
+}
+```
+
+2. For Next.js 13 App Directory, add `HighlightInit` to your `layout.tsx` file.
+
+```jsx
+// src/app/layout.tsx
+import './globals.css'
+
+import CONSTANTS from '../constants'
+import { HighlightInit } from '@highlight-run/next/client'
+
+export default function RootLayout({ children }: { children: React.ReactNode }) {
+	return (
+		<>
+			<HighlightInit
+				// excludedHostnames={['localhost']}
+				projectId={CONSTANTS.NEXT_PUBLIC_HIGHLIGHT_PROJECT_ID}
+				serviceName="my-nextjs-frontend"
+				tracingOrigins
+				networkRecording={{
+					enabled: true,
+					recordHeadersAndBody: true
+				}}
+			/>
+
+			<html lang="en">
+				<body>{children}</body>
+			</html>
+		</>
+	)
+}
+```
+
+3. Optionally add a React [Error Boundary](https://react.dev/reference/react/Component#catching-rendering-errors-with-an-error-boundary).
+
+```jsx
+// src/app/components/error-boundary.tsx
+'use client'
+
+import { ErrorBoundary as HighlightErrorBoundary } from '@highlight-run/next/client'
+
+export function ErrorBoundary({ children }: { children: React.ReactNode }) {
+	const isLocalhost =
```

Comment:
should this logic just live in the error boundary?

Reply:
This is an example of custom logic. I'll get rid of it.